etl/steps/data/garden/lis/2023-01-18/shared.py

```diff
@@ -0,0 +1,332 @@
+        },
+        "hcexp": {
+            "name": "Total consumption",
+            "description": "This measure is related to <b>total consumption</b>, including that stemming from goods and services that have been purchased by the household, and goods ans services that have not been purchased, but either given to the household from somebody else, or self-produced (available as `hcexp` in the LIS dataset).",
+        },
+    }
+
+    # Details for each equivalence scale (equivalized with LIS scale and per capita)
+    equivalence_scales_dict = {
+        "eq": {
+            "name": "equivalized",
+            "description": "This measure of income or consumption is <b>equivalized</b>. 'Equivalized' in this case means that household income or consumption is divided by the LIS equivalence scale (squared root of the number of household members) to address for economies of scale in the household.",
+        },
+        "pc": {
+            "name": "per capita",
+            "description": "This measure of income or consumption is <b>per capita</b>, which means that household income or consumption is divided by the total number of household members.",
+        },
+    }
+
+    # Details for each relative poverty line
+    rel_dict = {40: "40% of dhi median", 50: "50% of dhi median", 60: "60% of dhi median"}
+
+    # Details for each absolute poverty line
+    abs_dict = {
+        100: "$1 a day",
+        215: "$2.15 a day",
+        365: "$3.65 a day",
+        685: "$6.85 a day",
+        1000: "$10 a day",
+        2000: "$20 a day",
+        3000: "$30 a day",
+        4000: "$40 a day",
+    }
+
+    # Details for naming each decile/percentile
+    pct_dict = {
+        10: {"decile10": "Poorest decile", "decile9": "Poorest decile"},
+        20: {"decile10": "2nd decile", "decile9": "2nd decile"},
+        30: {"decile10": "3rd decile", "decile9": "3rd decile"},
+        40: {"decile10": "4th decile", "decile9": "4th decile"},
+        50: {"decile10": "5th decile", "decile9": "5th decile"},
+        60: {"decile10": "6th decile", "decile9": "6th decile"},
+        70: {"decile10": "7th decile", "decile9": "7th decile"},
+        80: {"decile10": "8th decile", "decile9": "8th decile"},
+        90: {"decile10": "9th decile", "decile9": "Richest decile"},
+        99: {"decile10": "Top 1%", "decile9": "Top 1%"},
+        100: {"decile10": "Richest decile", "decile9": ""},
+    }
+
+    # To avoid breaking f-strings, I need to add a line break like this
+    new_line = "\n\n"
+
+    # Get a list of all the variables available
+    cols = list(tb_garden.columns)
+
+    for var, slug in var_slugs.items():
+        for wel in inc_cons_vars:
+            for e in equivalence_scales:
+
+                # For variables that use income variable and equivalence scale
+                col_name = f"{var}_{wel}_{e}"
+
+                if col_name in cols:
+
+                    tb_garden[
```

Comment:
Perhaps having a function returning Metadata would be more readable (and possibly testable), e.g.

```python
def var_metadata_income_and_equivalence_scale(slug, inc_cons, equivalence_scales) -> VariableMeta:
                      meta = VariableMeta(
                          title=f"{slug['title']} ({inc_cons['name']}, {equivalence_scales['name']})",
                          description=f"{slug['description']}{new_line}{inc_cons_dict['description']}{new_line}{equivalence_scales['description']}",
                          unit = slug["unit"],
                          short_unit = slug["short_unit"],
                      )
                      meta.display = {
                          "name": meta.title,
                          "numDecimalPlaces": slug["numDecimalPlaces"],
                      }
                      return meta

...

tb_garden[col_name].metadata = var_metadata_income_and_equivalence_scale(slug, inc_cons_dict[wel], equivalence_scales_dict[e])
```

This might be used to DRY similar code below.

Reply:
Thanks, I will think about this. It is convoluted because there are several variables that "behave" differently from each other, but this could work.